test/e2e/rollback_test.go

```diff
@@ -0,0 +1,54 @@
+package e2e
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	nmstatev1alpha1 "github.com/nmstate/kubernetes-nmstate/pkg/apis/nmstate/v1alpha1"
+)
+
+var _ = Describe("NodeNetworkState controller", func() {
+	var (
+		br1Absent = nmstatev1alpha1.State(`interfaces:
+  - name: br1
+    type: linux-bridge
+    state: absent
+`)
+
+		br1 = nmstatev1alpha1.State(`interfaces:
+  - name: br1
+    type: linux-bridge
+    state: up
+    bridge:
+      options:
+        stp:
+          enabled: false
+      port: []
+`)
+	)
+	Context("when there is an error at vlan-filtering", func() {
+		BeforeEach(func() {
+			By("Rename vlan-filtering to force failure after nmstatectl set")
+			runAtPods("sudo", "mv", "/usr/local/bin/vlan-filtering", "/usr/local/bin/vlan-filtering.bak")
+		})
+		AfterEach(func() {
+			By("Correct the vlan-filtering script name")
+			runAtPods("sudo", "mv", "/usr/local/bin/vlan-filtering.bak", "/usr/local/bin/vlan-filtering")
+			updateDesiredState(br1Absent)
+			for _, node := range nodes {
+				interfacesNameForNodeEventually(node).ShouldNot(ContainElement("br1"))
+			}
+		})
+		It("should rollback nmstate state to previous value", func() {
```

Comment:
"should rollback failed state configuration"

We should describe that we "fail the configuration" here, since it is not a part of the context description